patterns/ffi-wrappers.md

```diff
@@ -0,0 +1,106 @@
+# Type Consolidation into Wrappers
+
+## Description
+
+This pattern is designed to allow gracefully handling multiple related types, while minimizing the surface area for memory unsafety.
+
+One of the cornerstones of Rust's aliasing rules is lifetimes. This ensures many patterns of access between types can be memory safe, including against data races.
+
+However, when Rust types are exported to other languages, they are usually transformed into pointers. In Rust, a pointer means "the user manages the lifetime of the pointee." It is their responsibility to avoid memory unsafety.
```

Comment:
`Pointee` wasn't intended I assume?

```suggestion
However, when Rust types are exported to other languages, they are usually transformed into pointers. In Rust, a pointer means "the user manages the lifetime of the pointer." It is their responsibility to avoid memory unsafety/ensure memory safety.
```